src/Symfony/Component/HttpClient/Response/TraceableResponse.php

```diff
@@ -63,12 +63,12 @@ public function getContent(bool $throw = true): string
 
     public function toArray(bool $throw = true): array
     {
-        $this->content = $this->response->toArray(false);
```

Comment:
the intent of doing it this way way to populate $this->content even if checkStatusCode throws
we should preserve this behavior while fixing the precedence of exceptions

Reply:
is it actually valid to store the array in `$this->content` ? Wouldn't this break using `->getContent()` ?

Reply:
$this->content is never read here, it's always *written* to.

Reply:
OH btw, the same precedence issue would happen with getContent in case of a TransportException

Reply:
this is exactly my point. The value it writes does not correspond to the contract of `$response->getContent()`

Reply:
neither does getContent read $this->content :+1: 
$this->content is a pure collected info, for the data collector (and its type varies depending on how the code used the response object yes)